packages/docs/docs/auth/index.md

```diff
@@ -12,6 +12,22 @@ Authentication (_are you who you say you are?_) and authorization (_what can you
 
 Users are the representation of identities in Medplum, and each user belongs to one or more Medplum Projects. For a specific project, a user can be either a [Practitioner](../api/fhir/resources/practitioner.mdx), [Patient](../api/fhir/resources/patient.mdx) or [Bot](../bots/index.md). At a high level, Practitioners are staff or administrators, Patients are those receiving care and Bots are designed for programmatic access or integrations.
 
+## Login Flowchart
+
+Users can belong to multiple Medplum projects, and the service supports multiple types of authentication. Below is a diagram that steps through the login logic and process. There are three major stages in the login flow.
+
+| Stage       | Description                                                                                                                                                                                                                                                                                                | Relevant endpoints                         |
+| ----------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ------------------------------------------ |
+| Credentials | In the Credentials phase of login the authentication provider is determined (Medplum auth, Google auth, Okta, etc.), the credentials are collected and user authenticated with the auth provider of choice. If Okta is enabled for a domain, all emails at that domain will go to Okta for authentication. | `auth/login` <br /> `auth/external` <br /> |
+| Profile     | In the Profile phase, if the user is a member of multiple projects, one must be selected to proceed                                                                                                                                                                                                        | `auth/profile`<br />                       |
```

Comment:
@codyebberson I like this third column that outlines the endpoints, I think it makes it more concrete - but wanted your feedback.

Reply:
Not sure what the goal is of the section?  The "/auth/" endpoints are pseudo-internal endpoints at the moment.  I.e., they're an implementation detail, and not intended for customers to use directly.  No expectations of backwards compatibility, etc.

That is in contrast to the "/oauth/" endpoints, which are well defined with the OpenID/OAuth spec, which we are committed to following.

If necessary, then we could change that, and decide to make them official public API's.  If that's the case, then we would need more details.  I.e., list out each of the endpoints with inputs and outputs for each.

I also don't think this belongs on the top-level Auth page.  It is deep in the weeds, and (in my opinion) this page should be more about big picture of auth.

With regards to the actual content...

Suggestions:

- Split "Users can belong to multiple Medplum projects, and the service supports multiple types of authentication. " into two sentences.
- Let's move the diagram above the table?
- "There are three major stages in the login flow." - there are now 4 major stages if you include "Domain"
- `auth/google` could be added to the "Endpoints" column
- Other endpoints?
  - `auth/method` to determine auth methods for an email address
  - `auth/me` for details about current login
  - `auth/mfa` for 2FA input
  - `auth/newpatient` for new patient registration
  - `auth/newuser` for new user registration
  - `auth/newproject` for new project registration
  - etc

Overall.... I think we need to be clear and consistent about the intentions of the page.

Reply:
This is an [explanation](https://diataxis.fr/explanation/) style page, and so I'm taking cues off of that.  The endpoints, though an implementation detail, give people a framework to understand what's going on and introduce them to concepts that they will need to understand like "Project", "Profile" which are Medplum specific.  

This is challenging, and it is rare to see good documentation on this front, but I think that a peek under the hood here will help in terms of clarity.  Personally, browsing the auth endpoints was the most helpful for me in trying to get oriented.

Will add the "Domain" step and draft of additional endpoints.

Reply:
Your call.  My 2 cents is to remove the endpoints, because it adds cognitive overhead, and we do not provide any additional supporting information.